### DISCUSSION TITLE
Why is the audio padded with `N_SAMPLES` instead of `HOP_LENGTH`

### DISCUSSION THREAD

**Discussion by MahmoudAshraf97** [openai/whisper#2422] [Ideas]
Hello, I noticed that if we pad the audio with `HOP_LENGTH` instead of `N_SAMPLES` the resulting features should be identical since the extra padding does not contribute to the STFT calculation

```python
from whisper import log_mel_spectrogram
import torch

SAMPLING_RATE = 16000
N_SAMPLES = SAMPLING_RATE * 30
HOP_LENGTH = 160
N_FRAMES = N_SAMPLES // HOP_LENGTH


for i in range(100):
    audio_n_samples = torch.randint(low=1, high=30, size=(1,)).item() * SAMPLING_RATE

    audio = torch.rand(audio_n_samples)

    features_pad_hop_length = log_mel_spectrogram(audio, padding=HOP_LENGTH)

    features_pad_30s_zeros = log_mel_spectrogram(audio, padding=N_SAMPLES)

    assert torch.allclose(
        features_pad_hop_length[:, :-1],
        features_pad_30s_zeros[:, :-N_FRAMES],
    )
```

and the feature frames that correspond to the extra padding are cropped anyway here:
https://github.com/openai/whisper/blob/271445b2f24f00f8175c4fb7ae91876f7451dfc1/whisper/transcribe.py#L140
https://github.com/openai/whisper/blob/271445b2f24f00f8175c4fb7ae91876f7451dfc1/whisper/transcribe.py#L281-L282

I want to hear thoughts about whether this is a valid idea or not, I will open a PR if it is, this will make the feature extraction faster and will use less resources especially for short audios
